src/binary-reader.cc

```diff
@@ -432,8 +432,7 @@ Result BinaryReader::ReadOffset(Offset* offset, const char* desc) {
 Result BinaryReader::ReadAlignment(Address* alignment_log2, const char* desc) {
   uint32_t value;
   CHECK_RESULT(ReadU32Leb128(&value, desc));
-  if (value >= 128 ||
-      (value >= 32 && !options_.features.multi_memory_enabled())) {
+  if (value >= 128 || (value & ((1 << 6) - 1)) >= 32) {
```

Comment:
This is a bit gnarly... is there any way to simplify/factor this?

Reply:
this is the simplest approach we could find without duplicating the entire `ReadAlignment`. but um... yeah we don't like it either.

`ReadAlignment` is currently used in `ReadMemLocation` and in `ReadLinkingSection` (for `SegmentInfo`), and as far as we can tell the latter doesn't want multi-memory flag stuff. so its use is a bit overloaded... we could split them, we guess? thoughts?